app/controllers/bank_accounts_controller.rb

```diff
@@ -0,0 +1,53 @@
+class BankAccountsController < ApplicationController
+  def index
+    if current_user.banker ==true || current_user.staff == true
+      @accounts = BankAccount.all
+      characters = Character.all
+      @characters_without_accounts = []
+      characters.each do |character|
+        if !BankAccount.has_account?(character.id) == true && character.archived==false
+          @characters_without_accounts << character
+        end
+      end
+      @account = BankAccount.new
+    else
+      authenticate_staff!
+    end
+  end
+
+  def show
+    @character = Character.find_by(slug: params[:character_id])
+      if !@character
+        @character = Character.find(params[:character_id])
+      end
+    if !current_user.banker && !current_user.staff
+      authorize_user(@character.user)
+    end
+    @account = @character.bank_account
+    @transactions = @account.bank_transactions
+  end
+
+
+  def create
+    @account = BankAccount.new(account_params)
+    @character = Character.find(account_params[:character_id])
+
+    if @account.save
+      redirect_to character_bank_account_path(@character.id,@account.id)
```

Comment:
Two things, here. One, you don't need to call `.id` when passing `ActiveRecord` objects to the path helpers. Just `redirect_to character_bank_account_path(@character, @account)`.

Secondly: is this not always going to be the same as `character_bank_account_path(@account.character, @account)`? If it is, that would save you looking up the `@character` record at all -- remember, database queries are time-consuming.

Reply:
refactored per your suggestion